### PR TITLE
extract odk data

### PR DESCRIPTION
**Pull request by erichesse**
# PR Context

# Added
- add odk extract

# Changes

# Deprecated

# Removed

# Fixed

# Security

### CODE REVIEW

File: tests/odk/test_extract.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            "end_group",
+            nan,
+            "begin_group",
+            "select_one consent",
+            "note",
+            "select_one yes_no",
+            "end_group",
+        ],
+    }
+    assert test[0].model_dump(exclude_defaults=True) == expected
```
Comment (cutoffthetop) [inline tests/odk/test_extract.py:203]: small unittest for `get_column_dict_by_pattern` would be nice

File: mex/odk/model.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from mex.common.models import BaseModel
+
+
+class ODKData(BaseModel):
+    """Model class for odk data."""
+
+    file_name: str
+    hint: dict[str, list[str | float]]
+    label_choices: dict[str, list[str | float]]
+    label_survey: dict[str, list[str | float]]
```
Comment (cutoffthetop) [inline mex/odk/model.py:10]: why did you go with go with lists of column values instead of one model per row?

**Review by cutoffthetop — CHANGES_REQUESTED** [review]

File: assets/mappings/__final__/odk/resource_3rdWave-NAM.yaml [hunk 3]
```diff
@@ -0,0 +1,173 @@
+# yaml-language-server: $schema=../../__schema__/ExtractedResource_MappingSchema.json
```
[the remark this answers is not in the document]
Comment (erichesse) [reply assets/mappings/__final__/odk/resource_3rdWave-NAM.yaml]: I did this to get the test coverage of odk/main to over 95%.

I now redirected the files to test_mapping, so we don't need to maintain mappings for "every" x-system. Also made them a little more generic.

https://github.com/robert-koch-institut/mex-extractors/pull/10/commits/47a599c07a483fa458c22ce5c7c32e4466c42c22

File: tests/odk/test_extract.py (hunk 1 above)
Comment (erichesse) [reply tests/odk/test_extract.py:204]: https://github.com/robert-koch-institut/mex-extractors/pull/10/commits/8dc2471d941370d7953638751faeaee8ce525848

File: mex/odk/model.py (hunk 2 above)
Comment (erichesse) [reply mex/odk/model.py:10]: The basic data structure here are not rows, but rather groups marked by "begin_group" and "end_group" in the "type" column. 

To properly validate that, you would already need to do complex data transformation. But doing that in the extraction step would violate the "separation of concern" principle.

Comment (cutoffthetop) [reply mex/odk/model.py:10]: uff, i see. but i do agree that this is the way to go then

File: assets/mappings/__final__/odk/resource_3rdWave-NAM.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply assets/mappings/__final__/odk/resource_3rdWave-NAM.yaml]: alright, let's see how this works out. do you think we could do the same for the other mapping yamls? shouldn't be on this branch, just curious about your assessment and whether we should create a follow-up task.

one config-issue though; i would prefer setting the default in `ODKSettings` to the generic mappings and then pointing the setting in dev/stage/prod to the actual path. that way, the extractor would remain runnable in its default configuration even when the assets repository is not set up.

[the remark this answers is not in the document]
Comment (erichesse) [reply assets/mappings/__final__/odk/resource_3rdWave-NAM.yaml]: yes, this is a good idea. I created a ticket for that: MX-1570

I liked to also manually run e.g. `poetry run odk` with the actually mappings to see whether it works on top of tests. Althoug I can see how that is not best practices. Another issue is, that with that change the default value extraction will not be visible in the locally dumped ndjson but only in staging. Anyway I redirected the settings: https://github.com/robert-koch-institut/mex-extractors/pull/10/commits/3f5f4475fc6804e0a9a6f559009235729a4caf4a

**Review by cutoffthetop — APPROVED** [review]
lgtm ✨ :shipit:

**Pull request merged**